gslib/tests/test_stet_cp.py

```diff
@@ -0,0 +1,152 @@
+    # Make shell script executable in addition to current permissions.
+    current_stat = os.stat(self.stet_binary_path)
+    os.chmod(self.stet_binary_path, current_stat.st_mode | stat.S_IEXEC)
+    # Fake config.
+    self.stet_config_path = self.CreateTempFile()
+
+  def test_encrypts_upload_if_stet_is_enabled(self):
+    object_uri = self.CreateObject()
+    test_file = self.CreateTempFile(contents='will be rewritten')
+
+    stderr = self.RunGsUtil([
+        '-o', 'GSUtil:stet_binary_path={}'.format(self.stet_binary_path), '-o',
+        'GSUtil:stet_config_path={}'.format(
+            self.stet_config_path), 'cp', '--stet', test_file,
+        suri(object_uri)
+    ],
+                            return_stderr=True)
+
+    # Progress indicator should show transformed file size (variable based on
+    # random string generation above). 4.0 B is the pre-transform size.
+    self.assertNotIn('/4.0 B]', stderr)
+
+    stdout = self.RunGsUtil(['cat', suri(object_uri)], return_stdout=True)
+    self.assertIn('subcommand: encrypt', stdout)
+    self.assertIn('config file: --config-file={}'.format(self.stet_config_path),
+                  stdout)
+    self.assertIn('blob id: --blob-id={}'.format(suri(object_uri)), stdout)
+    self.assertIn('in file: {}'.format(test_file), stdout)
+    self.assertIn('out file: {}_.stet_tmp'.format(test_file), stdout)
+
+    self.assertFalse(
+        os.path.exists(
+            temporary_file_util.GetStetTempFileName(
+                storage_url.StorageUrlFromString(test_file))))
+
+  def test_decrypts_download_if_stet_is_enabled(self):
+    object_uri = self.CreateObject(contents='abc')
+    test_file = self.CreateTempFile()
+
+    stderr = self.RunGsUtil([
+        '-o', 'GSUtil:stet_binary_path={}'.format(self.stet_binary_path), '-o',
+        'GSUtil:stet_config_path={}'.format(
+            self.stet_config_path), 'cp', '--stet',
+        suri(object_uri), test_file
+    ],
+                            return_stderr=True)
+
+    # Progress indicator should show transformed file size (variable based on
+    # random string generation above). 4.0 B is the pre-transform size.
+    self.assertNotIn('/4.0 B]', stderr)
+
+    with open(test_file) as file_reader:
+      downloaded_text = file_reader.read()
+    self.assertIn('subcommand: decrypt', downloaded_text)
+    self.assertIn('config file: --config-file={}'.format(self.stet_config_path),
+                  downloaded_text)
+    self.assertIn('blob id: --blob-id={}'.format(suri(object_uri)),
+                  downloaded_text)
+    self.assertIn('in file: {}'.format(test_file), downloaded_text)
+    self.assertIn('out file: {}_.stet_tmp'.format(test_file), downloaded_text)
+
+    self.assertFalse(
+        os.path.exists(
+            temporary_file_util.GetStetTempFileName(
+                storage_url.StorageUrlFromString(test_file))))
+
+  def test_does_not_seek_ahead_for_bytes_if_stet_transform(self):
```

Comment:
kokoro seems to be failing for this

Reply:
Pushed +- 0.1 KiB regex, will wait for result

Reply:
Looks fine